applications/ChimeraApplication/python_scripts/navier_stokes_solver_vmsmonolithic_chimera.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.ChimeraApplication as KratosChimera
+# Import applications
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# Import base class file
+from navier_stokes_solver_vmsmonolithic import NavierStokesSolverMonolithic
+
+def CreateSolver(main_model_part, custom_settings):
+    return NavierStokesSolverMonolithicChimera(main_model_part, custom_settings)
+
+class NavierStokesSolverMonolithicChimera(NavierStokesSolverMonolithic):
+    def __init__(self, model, custom_settings):
+        super(NavierStokesSolverMonolithicChimera,self).__init__(model,custom_settings)
+
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesSolverMonolithicChimera", "Construction of NavierStokesSolverMonolithic finished.")
+
+    def AddVariables(self):
+        super(NavierStokesSolverMonolithicChimera,self).AddVariables()
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FLAG_VARIABLE)
+
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesSolverMonolithicChimera", "Fluid solver variables added correctly.")
+
+    def Initialize(self):
```

Comment:
@KratosMultiphysics/fluid-dynamics if the functions on the fluid-solver would be more fine-grained then this could be simplified a lot

Reply:
ping @KratosMultiphysics/fluid-dynamics

Reply:
ping @KratosMultiphysics/fluid-dynamics @rubenzorrilla 
I see this as critical for this PR.
I don't think it is much effort since we already have this concept working for years in StructuralMechanics
This can save us a lot of debugging in the future, and it has been on the list for long (e.g. #4390)
I can help, let me know

Reply:
Agree. Let me add it to the @KratosMultiphysics/fluid-dynamics project.

Reply:
I'll try to find the time to do it before we merge this so @adityaghantasala can take benefit of it.

Reply:
thanks a lot!